dimelo/load_processed.py

```diff
@@ -140,42 +196,44 @@ def pileup_vectors_from_bedmethyl(
     valid_base_counts = np.zeros(region_len, dtype=int)
     modified_base_counts = np.zeros(region_len, dtype=int)
     for chromosome, region_list in regions_dict.items():
-        for start_coord, end_coord, _ in region_list:
+        for start_coord, end_coord, strand in region_list:
             # TODO: This is not used anywhere; disabling for now
             # center_coord = (start_coord+end_coord)//2
             if chromosome in source_tabix.contigs:
                 for row in source_tabix.fetch(chromosome, start_coord, end_coord):
                     tabix_fields = row.split("\t")
                     pileup_basemod = tabix_fields[3]
+                    pileup_strand = tabix_fields[5]
                     keep_basemod = False
-                    if len(pileup_basemod.split(",")) == 3:
+                    if single_strand and pileup_strand.strip() != strand:
+                        # We are on the wrong strand, skip the rest of the steps for this row
+                        continue
+                    elif len(pileup_basemod.split(",")) == 3:
                         pileup_modname, pileup_motif, pileup_mod_coord = (
                             pileup_basemod.split(",")
                         )
                         if (
-                            pileup_motif == mod_motif
-                            and pileup_mod_coord == mod_coord_in_motif
-                            and pileup_modname
-                            in utils.BASEMOD_NAMES_DICT[
-                                mod_motif[int(mod_coord_in_motif)]
-                            ]
+                            pileup_motif == parsed_motif.motif_seq
+                            and int(pileup_mod_coord) == parsed_motif.modified_pos
+                            and pileup_modname in parsed_motif.mod_codes
                         ):
                             keep_basemod = True
                     elif len(pileup_basemod.split(",")) == 1:
-                        if (
-                            pileup_basemod
-                            in utils.BASEMOD_NAMES_DICT[
-                                mod_motif[int(mod_coord_in_motif)]
-                            ]
-                        ):
+                        if pileup_basemod in parsed_motif.mod_codes:
                             keep_basemod = True
                     else:
                         raise ValueError(
                             f"Unexpected format in bedmethyl file: {row} contains {pileup_basemod} which cannot be parsed."
                         )
                     if keep_basemod:
                         pileup_info = tabix_fields[9].split(" ")
-                        pileup_coord_relative = int(tabix_fields[1]) - start_coord
+                        if regions_5to3prime and strand == "-":
```

Comment:
If you assign `int(tabix_fields[1])` to a descriptive variable name before this if statement, it will drastically improve the readability of the coordinate transform operations below.

Reply:
Yeah we should probably do that sort of thing all over the place more than we do right now.